salt/utils/extend.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+'''
+SaltStack Extend
+~~~~~~~~~~~~~~~~
+
+A templating tool for extending SaltStack.
+
+Takes a template directory and merges it into a SaltStack source code
+directory. This tool uses Jinja2 for templating.
+
+This tool is accessed using `salt-extend`
+
+    :codeauthor: :email:`Anthony Shaw <anthonyshaw@apache.org>`
+'''
+from __future__ import absolute_import
+from __future__ import print_function
+import yaml
+from collections import OrderedDict
+from datetime import date
+import tempfile
+import os
+import shutil
+from jinja2 import Template
+
+# zip compat for PY2/3
+from salt.ext.six.moves import zip
+
+import logging
+log = logging.getLogger('salt-extend')
+
+try:
+    import click
+    HAS_CLICK = True
+except ImportError as ie:
+    HAS_CLICK = False
+
+
+def _get_template(path, option_key):
+    """
+    Get the contents of a template file and provide it as a module type
+
+    :param path: path to the template.yml file
+    :type  path: ``str``
+
+    :param option_key: The unique key of this template
+    :type  option_key: ``str``
+
+    :returns: Details about the template
+    :rtype: ``tuple``
+    """
+    with open(path, "r") as template_f:
+        template = yaml.load(template_f)
+        info = (option_key, template.get('description',''), template)
+    return info
+
+
+def _fetch_templates(src):
+    """
+    Fetch all of the templates in the src directory
+
+    :param src: The source path
+    :type  src: ``str``
+
+    :rtype: ``list`` of ``tuple``
+    :returns: ``list`` of ('key', 'description')
+    """
+    templates = []
+    log.debug('Listing contents of {0}'.format(src))
+    for item in os.listdir(src):
+        s = os.path.join(src, item)
+        if os.path.isdir(s):
+            template_path = os.path.join(s, 'template.yml')
+            if os.path.isfile(template_path):
+                try:
+                    templates.append(_get_template(template_path, item))
+                except:
```

Comment:
This is a bare `except` if you can't specify an exact exception, make pylint shut up about it by appending the following to this line:
```
# pylint disable=bare-except
```